common/predictive-text/worker/models/trie-model.ts

```diff
@@ -323,16 +339,38 @@
         } else {
           // When an entry is up next in the queue, we just add its contents to
           // the results!
-          results.push(next.content);
+          results.push({
+            text: next.content,
+            p: next.weight / N
+          });
           if (results.length >= limit) {
             return results;
           }
         }
       }
     }
     return results;
+
   }
-  
+
+  /**
+   * O(n) lookup to determine the total amount of nodes in the trie, starting
+   * at the provided node. Don't use this if you want lookups to be fast.
+   * @param node
+   */
+  function countNodes(node: Node): number {
```

Comment:
So, this adds up the weights from the leaves, correct?  Not the 'total amounts of nodes in the trie'?  I'm looking at example.str.trie_sencoten.model.js for comparison, and its weights are far higher than this comment would suggest.

Reply:
Good catch! That's correct; this should say it counts the weights of the leaves, and not how many nodes there are themselves.